python/version_label/version_label.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+from sgtk.platform.qt import QtCore, QtGui
+
+from .ui import resources_rc
+
+class VersionLabel(QtGui.QLabel):
+    """
+    Subclassed ``QLabel`` that displays a playback icon
+    centered above the existing pixmap.
+    
+    By populating an instance with shotgun version data
+    via the :any:`set_shotgun_data()` method, the version label
+    will look at the data and determine whether a playback 
+    icon should be displayed or not. In the case an icon is
+    displayed, a playback_clicked signal may be emitted. 
+    
+    :signal playback_clicked(dict): The playback icon was clicked. 
+        This signal passes the shotgun version data specified in
+        via the :any:`set_shotgun_data()` method back
+        to the caller.   
+    """
+    
+    # signal fires when the play button was clicked
+    playback_clicked = QtCore.Signal(dict)
+    
+    def __init__(self, parent):
+        """
+        Constructor
+        
+        :param parent: QT parent object
+        """
+        QtGui.QLabel.__init__(self, parent)
+        self._play_icon = QtGui.QPixmap(":/tk_framework_qtwidgets.version_label/play_icon.png")
+        self._play_icon_inactive = QtGui.QPixmap(":/tk_framework_qtwidgets.version_label/play_icon_inactive.png")
+        self._sg_data = None
+        self._hover = False
+        self._active = False
+
+    def set_shotgun_data(self, sg_data):
+        """
+        Sets version data associated with this label.
+        
+        :param sg_data: Shotgun version data, including quicktime fields.
+        """
+        self._sg_data = sg_data
+        
+        # based on the data, figure out if the icon should be active or not
+        self._active = False
+        
+        if sg_data and sg_data.get("type") == "Version":
+            # versions are supported
+            if sg_data.get("sg_uploaded_movie"):
+                self._active = True
```

Comment:
Since the logic for determining whether the data is playable is in this class, perhaps there should be a `playable` property exposed for use by the calling code. Having that property available would allow the user to selectively connect to the `playback_clicked` signal rather than assuming the **Version** data is playable. 

I'd probably recommend changing `_active` to `_playable` as well.